app/server.js

```diff
@@ -8,6 +8,222 @@ var uid = require('rand-token').uid;
 
 const PORT = 3001;
 
-http.createServer(function (request, response) {
+//api key storage
+const VALID_API_KEYS = ["98738203-8dsj-0983-9f72-di29dk38djk3", "8di3k99l-5305-8dk3-1849-zjdf938jfj98"];
+const TOKEN_VALIDITY_TIMEOUT = 15 * 60 * 10000;
 
-}).listen(PORT);
\ No newline at end of file
+//state-holding variables, cart is kept inside currentUser via /POST /api/me/cart
+let brands = [];
+let products = [];
+let users = [];
+let currentUser = {};
+let accessTokens = [];
+let failedLoginAttempts = {};
+
+//Setup for the router
+let myRouter = Router();
+myRouter.use(bodyParser.json());
+
+//Setup for server
+let server = http.createServer(function (request, response) {
+  if(!VALID_API_KEYS.includes(request.headers["x-authentication"])) {
+    response.writeHead(401, "You need to have a valid API key to use this API")
```

Comment:
This does check the API key, but it's not preventing me from accessing the API if I don't have a key. I get 200 (success) but with "You need to have a valid API key to use this API".